app/controllers/likes_controller.rb

```diff
@@ -1,31 +1,37 @@
 class LikesController < ApplicationController
 
  def index
-    # if current_user
     if params[:keyword] || params[:category_id] 
       permitted_params = params.permit(:category_id)
-      @pins = current_user.pins.includes(:category).where(permitted_params).where("title LIKE '%#{params[:keyword]}%'").order('created_at DESC').limit(params[:limit]).offset(params[:offset])
+      @pins = current_user.pins.includes(:category).where(permitted_params).where("title LIKE '%#{params[:keyword]}%'").order('likes.id DESC').limit(params[:limit]).offset(params[:offset])
     else
-      @pins = current_user.pins.includes(:category).order('created_at DESC').limit(params[:limit]).offset(params[:offset])
+      @pins = current_user.pins.includes(:category).order('likes.id DESC').limit(params[:limit]).offset(params[:offset])
     end
-      # @likes = current_user.likes.includes(:pin).limit(params[:limit]).offset(params[:offset])
-    # else
-      # @pins = nil
-    # end
   end
 
   def create
-    like = Like.new(permitted_params)
+    like = current_user.likes.new
+    like.pin_id = params[:pin_id]
     if like.save
-      redirect_to :back
+      render json: {success: true}
     else
-      puts "ERROR SAVING OUR LIKE "
+      render json: {success: false}
     end
   end
 
+  def destroy
+    like = current_user.likes.find_by(search_params)
+    like.destroy
+    render json: {status: 200, params: params}
+  end
+
   protected
   def permitted_params 
-    params.require(:like).permit(:user_id, :pin_id)
+    params.permit(:pin_id)
+  end
+
+  def search_params 
```

Comment:
I don't understand...? I changed it because I no longer require like or user info